operator/pkg/ciliumendpointslice/endpointslice.go

```diff
@@ -0,0 +1,329 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointslice
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/operator/metrics"
+	operatorOption "github.com/cilium/cilium/operator/option"
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	capi_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	csv2 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2"
+	csv2a1 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/informer"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	// cesNamePrefix is the prefix name added for the CiliumEndpointSlice
+	// resource.
+	cesNamePrefix = "ces"
+
+	// defaultSyncBackOff is the default backoff period for cesSync calls.
+	defaultSyncBackOff = 1 * time.Second
+	// maxSyncBackOff is the max backoff period for cesSync calls.
+	maxSyncBackOff = 100 * time.Second
+	// maxRetries is the number of times a cesSync will be retried before it is
+	// dropped out of the queue.
+	maxRetries = 15
+	// CEPs are batched into a CES, based on its Identity
+	cesIdentityBasedSlicing = "cesSliceModeIdentity"
+	// default qps limit value for workqueues, this only for retries.
+	CESControllerWorkQueueQPSLimit = 10
```

Comment:
Can you remind me have we tested 10?

Reply:
I think this QPS limit for workqueues, Yes we tested with 10.